src/permissions/Permissions.service.ts

```diff
@@ -116,7 +116,7 @@ export class PermissionsService {
 
     if (!hasRequiredRole) {
       this.logger.debug('User does not have one of the required roles. Returning an empty list of responsibilities. ');
-      return [];
+      return new FacilityAccessWithCertStatementFlagDTO();
```

Comment:
If I am not mistaken, plantList and missingCertificationStatements are not initialized in FacilityAccessWithCertStatementFlagDTO.  So, they will have 'undefined' as values when you create a new FacilityAccessWithCertStatementFlagDTO object and return it (which is unpredictable for downstream code).   Can we initialize at least those values and return? 

Perhaps something like this?

return {
    plantList: [],
    missingCertificationStatements: true,
  } as FacilityAccessWithCertStatementFlagDTO;

Also check other uses of new FacilityAccessWithCertStatementFlagDTO ...